src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -393,6 +392,36 @@ public class ShadeMojo extends AbstractMojo {
     @Parameter(defaultValue = "false")
     private boolean skip;
 
+    /**
+     * Extra JAR files to infuse into shaded result.
+     * <p>
+     * One can add here "extra JARs", to be worked into the resulting shaded JAR. The listed JAR files must exist.
+     * Extra JARs will be processed in same way as main JAR (if any): applied relocation, resource transformers
+     * but not filtering.
+     * <p>
+     * Note: this feature should be used lightly, is not meant as ability to replace dependency hull! It is more
+     * just a feature to be able to slightly "differentiate" shaded JAR from main only.
+     *
+     * @since 3.6.0
+     */
+    @Parameter
+    private List<File> extraJars;
+
+    /**
+     * Extra Artifacts to infuse into shaded result.
+     * <p>
+     * One can add here "extra Artifacts" to be worked into the resulting shaded JAR. The artifacts will be resolved
+     * (not transitively), and will be processed in same way as dependency JARs (if any): regarding relocation,
+     * resource transformers and filtering.
+     * <p>
+     * Note: this feature should be used lightly, is not meant as ability to replace dependency hull! It is more
+     * just a feature to be able to slightly "differentiate" shaded JAR from main only.
+     *
+     * @since 3.6.0
+     */
+    @Parameter
+    private List<String> extraArtifacts;
```

Comment:
We should describe format of item ... `groupId:artifactId:...` 
which items are required?

Reply:
The ["standard" format](https://github.com/apache/maven-resolver/blob/master/maven-resolver-api/src/main/java/org/eclipse/aether/artifact/DefaultArtifact.java#L51-L63), will add to comment.